app/jobs/regular/export_csv_file.rb

```diff
@@ -87,7 +87,12 @@ def execute(args)
       end
 
     ensure
-      notify_user(download_link, file_name, file_size, export_title)
+      post = notify_user(download_link, file_name, file_size, export_title)
+      if user_export.present? && post.present?
+        topic = post.topic
+        user_export.update_columns(topic_id: topic.id)
+        topic.update_status('closed', true, Discourse.system_user)
```

Comment:
I'm not sure about this but are closed PMs deleted eventually?

Reply:
No. I am closing the PM here per Sam's request. For hard deleting the PM I added `belongs_to :topic, dependent: :destroy` in `user_export.rb`.

Reply:
I think destroying a topic doesn't destroy it posts.

Reply:
Fixed.